lib/service_base_url_test_kit/service_base_url_validate_group.rb

```diff
@@ -150,15 +162,20 @@ def find_referenced_endpoint(bundle_resource, endpoint_id_ref)
 
         skip_if bundle_resource.entry.empty?, 'The given Bundle does not contain any resources'
 
-        bundle_resource
+        endpoint_list = bundle_resource
           .entry
           .map(&:resource)
           .select { |resource| resource.resourceType == 'Endpoint' }
           .map(&:address)
           .uniq
-          .each do |address|
+
+        check_limit = get_endpoint_availability_limit(endpoint_availability_limit)
+
+        endpoint_list.each_with_index do |address, index|
           assert_valid_http_uri(address)
 
+          next if check_limit.present? && index >= check_limit
```

Comment:
Would it be a bit more readible and consice to just do something like this:

`next if endpoint_availability_limit.present? && endpoint_availability_limit.to_i <= index`

I'm not sure what value you get out of that `get_` method and hte extra `check_limit` variable.  Fortunately `.to_i` seems to be tolerant to wierd inputs like `a` or `1a`, so long as it isn't null (which it can't be to get past `present?`).